### PR TITLE
Feat: multiple templates rewrites plp

### PR DESCRIPTION
**Pull request by eduardoformiga**
#### Types of changes
- [x] New content (guides, endpoints, app documentation)
- [ ] Improvement (make a documentation even better)
- [ ] Fix (fix a documentation error)
- [ ] Spelling and grammar accuracy (self-explanatory)

This PR aims to add docs regarding how to use the rewrites feature along with the Multiple page templates in PLP for FastStore.

FastStore PR
- https://github.com/vtex/faststore/pull/2320

### CODE REVIEW

File: docs/faststore/docs/headless-cms/multiple-page-template.mdx [hunk 1]
```diff
@@ -94,14 +95,58 @@ For example:
...
+    afterFiles: [],
+    fallback: [],
+  }
+},
+```
+
+In the example above, the new `source` route `/my-office` will be rewritten to the existing `destination` route `/office`, using the `/office` template without changing the URL.
+In that way, you can use the existing `destination` route `/office` to map this template in the hCMS.
+
+From the version `3.0.X` of the `@faststore/core` package, it's also possible to use the `source` route in the hCMS to map the template.
```
Comment (eduardoformiga) [inline docs/faststore/docs/headless-cms/multiple-page-template.mdx:134]: 🎗️ update the version after FastStore generate the new version

File: docs/faststore/docs/headless-cms/multiple-page-template.mdx [hunk 2]
```diff
@@ -94,14 +95,58 @@ For example:
...
+
+Update your store’s `@faststore/core` package to the `3.0.X` by running the following command:
+
+```bash
+yarn add @faststore/core@3.0.X
```
Comment (eduardoformiga) [inline docs/faststore/docs/headless-cms/multiple-page-template.mdx:139]: 🎗️ update the version after FastStore generate the new version

Comment (eduardoformiga) [inline docs/faststore/docs/headless-cms/multiple-page-template.mdx:139]:
```suggestion
From the version `3.0.61` of the `@faststore/core` package, it's also possible to use the `source` route in the hCMS to map the template.

Update your store’s `@faststore/core` package to the `3.0.61` by running the following command:

```bash
yarn add @faststore/core@3.0.61
```

File: docs/faststore/docs/headless-cms/multiple-page-template.mdx [hunk 3]
```diff
@@ -94,14 +95,58 @@ For example:
...
+      },
+    ],
+    afterFiles: [],
+    fallback: [],
+  }
+},
+```
+
+In the example above, the new `source` route `/my-office` will be rewritten to the existing `destination` route `/office`, using the `/office` template without changing the URL.
+In that way, you can use the existing `destination` route `/office` to map this template in the hCMS.
```
Comment (Mariana-Caetano) [inline docs/faststore/docs/headless-cms/multiple-page-template.mdx:132]:
```suggestion
In that way, you can use the existing `destination` route `/office` to map this template in the [Headless CMS](https://developers.vtex.com/docs/guides/faststore/headless-cms-overview).
```

File: docs/faststore/docs/headless-cms/multiple-page-template.mdx [hunk 4]
```diff
@@ -94,14 +95,58 @@ For example:
...
+
+From the version `3.0.61` of the `@faststore/core` package, it's also possible to use the `source` route in the hCMS to map the template.
```
Comment (Mariana-Caetano) [inline docs/faststore/docs/headless-cms/multiple-page-template.mdx:134]:
```suggestion
From the version `3.0.61` of the `@faststore/core` package, it's also possible to use the `source` route in the Headless CMS to map the template. To update your store's `@faststore/core` package to the `3.0.61`, run the following command:
```

File: docs/faststore/docs/headless-cms/multiple-page-template.mdx [hunk 5]
```diff
@@ -94,14 +95,58 @@ For example:
...
+
+Update your store’s `@faststore/core` package to the `3.0.61` by running the following command:
```
Comment (Mariana-Caetano) [inline docs/faststore/docs/headless-cms/multiple-page-template.mdx:136]:
```suggestion
```

File: docs/faststore/docs/headless-cms/multiple-page-template.mdx [hunk 6]
```diff
@@ -94,14 +95,58 @@ For example:
...
+
+```bash
+yarn add @faststore/core@3.0.61
+```
```
Comment (Mariana-Caetano) [inline docs/faststore/docs/headless-cms/multiple-page-template.mdx:140]: Após esse comando, devo rodar `yarn dev` ?

File: docs/faststore/docs/headless-cms/multiple-page-template.mdx [hunk 7]
```diff
@@ -94,14 +95,58 @@ For example:
 - Department template: `/office` alternative for products within a department, ignoring category.
 - Generic PLP template: Used if no other matches are found.
 
+#### Using Rewrites with PLP template
+
+FastStore enables the use of [rewrites from Next.js](https://nextjs.org/docs/app/api-reference/next-config-js/rewrites) to create a custom route.
+It's possible to enable the multiple page template feature for the custom route as well by adding your rewrites function into the `faststore.config.js` file as the Next.js doc suggests:
```
Comment (Mariana-Caetano) [inline docs/faststore/docs/headless-cms/multiple-page-template.mdx:101]:
```suggestion
FastStore allows you to create custom routes using [Next.js rewrites](https://nextjs.org/docs/app/api-reference/next-config-js/rewrites).
It's possible to enable the Multiple page template feature for the custom route as well by adding your rewrites function into the `faststore.config.js` file as suggested in the [Next.js documentation](https://nextjs.org/docs/app/api-reference/next-config-js/rewrites):
```

File: docs/faststore/docs/headless-cms/multiple-page-template.mdx [hunk 8]
```diff
@@ -94,14 +95,58 @@ For example:
...
+
```
Comment (Mariana-Caetano) [inline docs/faststore/docs/headless-cms/multiple-page-template.mdx:102]:
```suggestion
1. Add your rewrite function to faststore.config.js:

```

File: docs/faststore/docs/headless-cms/multiple-page-template.mdx [hunk 9]
```diff
@@ -94,14 +95,58 @@ For example:
...
+  return [
+    {
+      source: '/my-office',
+      destination: `/office`,
+    },
+  ]
+},
+```
+
+or
```
Comment (Mariana-Caetano) [inline docs/faststore/docs/headless-cms/multiple-page-template.mdx:114]:
```suggestion
2. Alternatively, use the extended format:
```

File: docs/faststore/docs/headless-cms/multiple-page-template.mdx [hunk 10]
```diff
@@ -94,14 +95,58 @@ For example:
...
+async rewrites() {
+  return [
+    {
+      source: '/my-office',
+      destination: `/office`,
+    },
+  ]
+},
+```
+
```
Comment (Mariana-Caetano) [inline docs/faststore/docs/headless-cms/multiple-page-template.mdx:113]:
```suggestion

This rewrites the URL `/my-office` to the existing route `/office`, effectively using the `/office` template without modifying the displayed URL. This allows you to map the template in the Headless CMS to the `/office` route.

```

File: docs/faststore/docs/headless-cms/multiple-page-template.mdx (hunk 3 above)
Comment (Mariana-Caetano) [inline docs/faststore/docs/headless-cms/multiple-page-template.mdx:132]:
```suggestion
This format offers more flexibility for advanced rewrite scenarios.
```

Comment (Mariana-Caetano) [reply docs/faststore/docs/headless-cms/multiple-page-template.mdx:132]: Maybe this sentence for the other code?

**Review by Mariana-Caetano — CHANGES_REQUESTED** [review]

File: docs/faststore/docs/headless-cms/multiple-page-template.mdx (hunk 6 above)
Comment (eduardoformiga) [reply docs/faststore/docs/headless-cms/multiple-page-template.mdx:139]: yes, but this is implicit if you are using local development.

**Review by Mariana-Caetano — APPROVED** [review]

File: docs/faststore/docs/headless-cms/multiple-page-template.mdx [hunk 11]
```diff
@@ -94,14 +95,60 @@ For example:
...
+      },
+    ],
+    afterFiles: [],
+    fallback: [],
+  }
+},
+```
+
+In the example above, the new `source` route `/my-office` will be rewritten to the existing `destination` route `/office`, using the `/office` template without changing the URL.
+In that way, you can use the existing `destination` route `/office` to map this template in the [Headless CMS](https://developers.vtex.com/docs/guides/faststore/headless-cms-overview).
```
Comment (Mariana-Caetano) [inline docs/faststore/docs/headless-cms/multiple-page-template.mdx:135]:
```suggestion
This format offers more flexibility for advanced rewrite scenarios.
```

**Pull request merged**